lib/utility/logger.rb

```diff
@@ -20,7 +21,7 @@ def level=(log_level)
       end
 
       def logger
-        @logger ||= ::Logger.new(STDOUT)
+        @logger ||= Settings[:ecs_logging] ? EcsLogging::Logger.new(STDOUT) : ::Logger.new(STDOUT)
```

Comment:
I face issue if I use `App::Config` instead of `Settings` here:
```
/lib/app/config.rb:57:in `ent_search_es_config': uninitialized constant Utility::Logger (NameError)
```
I guess it's because there is a circular dependency between `App::Config` and `Utility::Logger`.

I believe we purposely designed it so that `App::Config` is only accessed by code under `App` module. Shouldn't the config be globally available?

The `Settings` object is [made available globally](https://github.com/rubyconfig/config#accessing-the-settings-object), and we can stop other code from accessing it.

cc @elastic/ingestion-team

Reply:
It's a chicken-and-egg problem that doesn't have a good way to be resolved IMO.

App::Config provides settings that are required for `Utility::Logger` initialization, such as `Logger.level` for example, but also uses `Utility::Logger` to log things.

We already do some logger init in https://github.com/elastic/connectors-ruby/blob/main/lib/utility/environment.rb#L16, so you can probably put the logic that instantiates the logger there.

I think also we need to remove usage of Logger from `App::Config` class, at least for now until we find a better way to initialise cross-dependent classes on startup.

Reply:
>We already do some logger init in https://github.com/elastic/connectors-ruby/blob/main/lib/utility/environment.rb#L16, so you can probably put the logic that instantiates the logger there.

Yes, considered. But Utility::Logger is a singleton class and there's no `initialize` method (I don't plan to add one).

> I think also we need to remove usage of Logger from `App::Config` class, at least for now until we find a better way to initialise cross-dependent classes on startup.

If the `ent-search` elasticsearch config overriding logic is still there, the usage of Logger from `App::Config` is absolutely necessary. Instead of removing `Logger`, I'm seriously considering removing the overriding logic. See my slack proposal [here](https://elastic.slack.com/archives/C03GAPTQ2JF/p1662704552916849).

Reply:
I honestly feel that we should move away from logger being a singleton and instantiate it when needed. App::Config can instantiate a regular one when doing initialization, workers will instantiate another one and pass to the classes that need it.

Reply:
@artem-shelkovnikov , let's have a separate discussion (and a separate PR) on how to refactor the logger. Any concern if I keep the usage of `Settings` here? Or any more comments on the ECS logging itself?

Reply:
Absolutely, let's do it in another PR!

I also see there's an ongoing discussion about the format of logs that I don't have much to add to, so my concerns were addressed, thanks!